src/AuthModule/components/ChangPass/ChangPass.tsx

```diff
@@ -0,0 +1,195 @@
+  };
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    setSpinner(true);
+    console.log(data);
+    const token = localStorage.getItem("adminToken");
+    if (!token) {
+      throw new Error("User is not authenticated");
+    }
+    try {
+      const response = await axios.put(
+        "https://upskilling-egypt.com:3003/api/v1/Users/ChangePassword",
+        data,
+        {
+          headers: {
+            Authorization: token,
+          },
+        }
+      );
+      console.log(response)
+      handleClose();
+      showSuccessToast("Password has been updated successfully");
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response) {
+        showErrorToast(error.response.data.message);
+      }
+    } finally {
+      setSpinner(false);
+    }
+  };
+
+  return (
+    <>
+      <div className=" row justify-content-center align-items-center">
+        <div className=" col-md-12">
+          <div className="  rounded-3 px-5 py-4">
+            <small className=" text-white">welcome to PMS</small>
+
+            <h4 className={`${styleChangPass.changeWord} mb-5 `}>
+              {" "}
+              Change Password
+              <div className="color "></div>
+            </h4>
+
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <div className="form-group conInput  mb-3">
+                <label className=" color-text" htmlFor="old ">
+                  Old Password
+                </label>
+                <input
+                  className={`${styleChangPass.styleInput} form-control mt-2 `}
+                  type={showPassword ? "text" : "password"}
+                  placeholder="Enter your Old Password  "
+                  {...register("oldPassword", {
+                    required: "old Password is required ",
+                  })}
+                  id="old"
+                />
+                <i
+                  className={`fa-regular fa-eye${showPassword ? "-slash" : ""}`}
+                  onClick={togglePasswordVisibility}
+                ></i>
+
+                {errors.oldPassword && (
+                  <div className="alert alert-danger  d-inline-block w-100 mt-1">
+                    {errors.oldPassword.message}
+                  </div>
+                )}
+                <div className="border_bottom"></div>
+              </div>
+
+              {/*new password */}
+
+              <div className="form-group conInput  my-4">
+                <label className=" color-text" htmlFor="New ">
+                  New Password
+                </label>
+                <input
+                  className={`${styleChangPass.styleInput} form-control mt-2 `}
+                  placeholder="Enter your New Password"
+                  type={showNewPassword ? "text" : "password"}
+                  {...register("newPassword", {
+                    required: "new Password is required",
```

Comment:
I suggest moving all the validation logic into a separate file and then reusing it in all forms like [this one. ](https://github.com/AnasKhaled7/PMS-React/blob/main/src/lib/InputValidator.ts)